scipy/stats/stats.py

```diff
@@ -6249,10 +6250,13 @@ def _calculate_winsorized_variance(a, g, axis):
 
 def _broadcast_concatenate(xs, axis):
     """Concatenate arrays along an axis with broadcasting."""
+    # prepend 1s to array shapes as needed
```

Comment:
This should be in a separate PR and possibly backported.

* Concatenating along positive `axis` could be considered ambiguous when arrays have different numbers of dimensions. This resolves the ambiguity by prepending 1s as needed to array shapes with different numbers of dimensions before doing anything else.
* Also, the `np.broadcast` line doesn't work when the array has zero size along `axis`. To fix this, just work with the array shapes themselves.